tests/test_sequences_and_iterators.cpp

```diff
@@ -93,6 +117,23 @@ TEST_SUBMODULE(sequences_and_iterators, m) {
         })
         ;
 
+    py::class_<Thing>(m, "Thing")
+        .def(py::init<>())
+        .def("__getitem__",
+                [](Thing &t, py::slice slice) -> py::list {
+                    return get_item_list(t, slice);
+                }
+        )
+        .def("reverse",
+                [](Thing &t) -> py::list {
+                    py::slice ordinary_indices(0, static_cast<ssize_t>(t.size()), 1);
```

Comment:
nit Intent for this line is a bit unclear.

My read is that it's testing construction options of `py::slice`. If so, I'd suggest just returning `py::slice` directly, and test it via Python directly?

Reply:
(Will include in suggested change push)